src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2022 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {useTable, Column} from "react-table";
+import {Spinner, SpinnerSize} from "@dellstorage/clarity-react/spinner/Spinner";
+
+/**
+ * General component description :
+ * DataGridWithInfiniteScroll :
+ * Datagrids are for organizing large volumes of data that users can perform actions on.
+ */
+
+/**
+ * @param {rows} rows data
+ * @param {columns} column details
+ * @param {className} CSS class names
+ * @param {style} to add custom CSS styles
+ * @param {isLoading} if true then show loading icon before table is rendered
+ */
+type DataGridProps = {
+    rows: {[key: string]: any}[];
+    columns: Column<{[key: string]: any}>[];
+    className?: string;
```

Comment:
We are not using className prop. Please use it in code along with style. How can user pass custom CSS classes for datagrid ?

Reply:
We are not using className prop as of now for basic dataGrid, plan is to include in the next feature, but since you mentioned in previous comment to add className prop, hence added it.
![image](https://user-images.githubusercontent.com/70266574/190324591-a019ddf0-5057-4885-8649-0c4589a611a3.png)

Followed clarity react for this, they have also not used it in the code - https://github.com/EMCECS/clarity-react/blob/master/src/datagrid/DataGrid.stories.tsx